### PR TITLE
feat(networks): add Linea network

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This adds support for Linea network

- [x] deploy contracts
- [x] verify lock proxy
- [x] deploy subgraph

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/networks/src/networks/linea.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    factoryAddress: '',
+    oracle: '',
+    quoterAddress: '',
+    universalRouterAddress: '',
+  },
+  universalCard: {
+    cardPurchaserAddress: '',
+    stripeDestinationCurrency: 'usdc',
+    stripeDestinationNetwork: '',
+  },
```
Comment (julien51) [inline packages/networks/src/networks/linea.ts:85]:
```suggestion
  uniswapV3: {
  },
  universalCard: {
  },
```

**Review by julien51 — APPROVED** [review]

**Comment by julien51** [comment]
This is blocked by our inability to deploy a subgraph for Linea.

**Comment by julien51** [comment]
@clemsos They have added my IP to their allow list. What do I need to do to deploy this subgraph?

**Comment by clemsos** [comment]
subgraph is live at https://graph-query.linea.build/subgraphs/name/unlock-protocol/graphql?query=%0A++++%23%0A++++%23+Welcome+to+The+GraphiQL%0A++++%23%0A++++%23+GraphiQL+is+an+in-browser+tool+for+writing%2C+validating%2C+and%0A++++%23+testing+GraphQL+queries.%0A++++%23%0A++++%23+Type+queries+into+this+side+of+the+screen%2C+and+you+will+see+intelligent%0A++++%23+typeaheads+aware+of+the+current+GraphQL+type+schema+and+live+syntax+and%0A++++%23+validation+errors+highlighted+within+the+text.%0A++++%23%0A++++%23+GraphQL+queries+typically+start+with+a+%22%7B%22+character.+Lines+that+start%0A++++%23+with+a+%23+are+ignored.%0A++++%23%0A++++%23+An+example+GraphQL+query+might+look+like%3A%0A++++%23%0A++++%23+++++%7B%0A++++%23+++++++field%28arg%3A+%22value%22%29+%7B%0A++++%23+++++++++subField%0A++++%23+++++++%7D%0A++++%23+++++%7D%0A++++%23%0A++++%23+Keyboard+shortcuts%3A%0A++++%23%0A++++%23++Prettify+Query%3A++Shift-Ctrl-P+%28or+press+the+prettify+button+above%29%0A++++%23%0A++++%23+++++Merge+Query%3A++Shift-Ctrl-M+%28or+press+the+merge+button+above%29%0A++++%23%0A++++%23+++++++Run+Query%3A++Ctrl-Enter+%28or+press+the+play+button+above%29%0A++++%23%0A++++%23+++Auto+Complete%3A++Ctrl-Space+%28or+just+start+typing%29%0A++++%23%0A++%7B%0A++++locks+%7B%0A++++++id%0A++++%7D%0A++%7D

**Comment by clemsos** [comment]
should prevent deploying from CI if IP is ont right

**Comment by julien51** [comment]
Adding the keyManager is failing:

```
✗ yarn run hardhat deploy:keyManager --locksmiths 0x22c095c69c38b66afAad4eFd4280D94Ec9D12f4C,0x903073735Bb6FDB802bd3CDD3b3a2b00C36Bc2A9 --network linea
An unexpected error occurred:

ProviderError: invalid opcode: PUSH0
    at HttpProvider.request (/Users/julien/repos/unlock/node_modules/hardhat/src/internal/core/providers/http.ts:88:21)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at EthersProviderWrapper.send (/Users/julien/repos/unlock/packages/hardhat-plugin/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
```

File: packages/networks/src/networks/linea.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  id: 59144,
+
+  isTestNetwork: false,
+
+  keyManagerAddress: '0x338b1f296217485bf4df6CE9f93ab4C73F72b57D',
+
+  maxFreeClaimCost: 10,
+
+  // multisig: '', // TODO
```
Comment (julien51) [inline packages/networks/src/networks/linea.ts:54]: @clemsos Please transfer to `unlock-protocol.eth` !

File: packages/networks/src/networks/linea.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  name: 'Linea',
+
+  nativeCurrency: {
+    coingecko: 'linea-eth',
+    decimals: 18,
+    name: 'Linea Ether',
+    symbol: 'ETH',
+  },
+  previousDeploys: [],
+  provider: 'https://rpc.linea.build',
```
Comment (julien51) [inline packages/networks/src/networks/linea.ts:65]: HA! I think we need to deploy this on our provider infra as well!

**Review by julien51 — COMMENTED** [review]
I need to set the Linea in our provider!

**Pull request merged**